tests/unit/EntityId/PrefixMappingEntityIdParserTest.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+namespace Wikibase\DataModel\Services\Tests\EntityId;
+
+use Wikibase\DataModel\Entity\EntityIdParser;
+use Wikibase\DataModel\Entity\EntityIdParsingException;
+use Wikibase\DataModel\Entity\EntityId;
+use Wikibase\DataModel\Entity\ItemId;
+use Wikibase\DataModel\Entity\PropertyId;
+use Wikibase\DataModel\Services\EntityId\PrefixMappingEntityIdParser;
+use Wikimedia\Assert\ParameterAssertionException;
+use Wikimedia\Assert\ParameterElementTypeException;
+use Wikimedia\Assert\ParameterTypeException;
+
+/**
+ * @covers Wikibase\DataModel\Services\EntityId\PrefixMappingEntityIdParser
+ *
+ * @license GPL-2.0+
+ */
+class PrefixMappingEntityIdParserTest extends \PHPUnit_Framework_TestCase {
+
+	private function getRegularParser( $input, EntityId $returnValue ) {
+		$regularParser = $this->getMock( EntityIdParser::class );
```

Comment:
I've apparently gone too far with mocking things here. So mocks are now only use with tests that don't really "parse" anything, in cases like here BasicEntityIdParser is used